### ISSUE TITLE
OTRS Import after LDAP import capitalization problem

### ISSUE THREAD

**Issue by smalchow** [zammad/zammad#1674]
### Infos:

* Used Zammad version: zammad-2.1.0
* Used Zammad installation source: rpm package
* Operating system: CentOS 7
* Browser + version: 

### Expected behavior:

Users from an OTRS import should be merged with a previous LDAP import.

### Actual behavior:

The users seem to be partly updated, but not all, specifically it looks as if the LDAP import converts everything to lowercase, where as the OTRS import keeps the login as is.

We can see that users with userPrincipalName such as "Givenname.Familyname@mail.com" are created, whereas "givenname.familyname@mail.com" is updated. The OTRS user in the first however is still created with a lowercase login. 

Also i can see that in the first case there is a 3 digit number appended to the login, such as

OTRS: Givenname.Familyname@mail.com -> zammad: givenname.familyname@mail.com123

This behaviour is the same for all users.

### Steps to reproduce the behavior:

- Use one directory with a few users like Givenname.Familyname@mail.com and some like givenname.familyname@mail.com
- import LDAP users first (in our case with changed order of "samaccountname" and "userprincipalname" in "lib/ldap/user.rb:61"
- then import the tickets and users from OTRS as described in the documentation.
- check the logfiles, the users with Givenname.Familyname@mail.com are like this

_thread#-: add User.find_by(login: Ggggg.Fffff@mail.com)_

the users with all lowercase are like this

_thread#-: update User.find_by(login: ggggg.fffff@mail.com)_ 

when i search for the first user in zammad using

_User.find_by(email: 'Ggggg.Fffff@mail.com')_

i do not find the user, if i search with lowercase

_User.find_by(email: 'ggggg.fffff@mail.com')_

i find him, but the login is like this

login: "ggggg.fffff@mail.com507"

**Comment by smalchow** [comment]
Maybe i solved the problem, (at least for me)

I changed 2 files in lib/import/otrs

user.rb
customer_user.rb

in each of the files I added this

        _mapped[:login].downcase!_

below line 70. (stricly not neccessary in the user.rb i think, but i am not a developer)

This should not be a problem with ldap backed installations, with other backends this could mean that the user has to login as "donald duck" instead of "Donald Duck" if i see this correctly.

so for now this is more of a dirty fix, but maybe someone with more development skills can make this better ?

**Comment by thorsteneckel** [comment]
Hi @smalchow - Thanks for your comprehensive analysis of the issue. In fact Zammad (tries to as we can see) downcases all identifiers (like e.g. the email) to avoid exactly this issue. So your solution is correct. This is a workaround for case sensitive databases to avoid custom / inperformant queries.

Would be great if you could create a pull request for this!

**Comment by monotek** [comment]
Ahhrrg.... This bugged me already 7 years ago :D 
https://bugs.otrs.org/show_bug.cgi?id=6063

**Comment by muhammadn** [comment]
I am writing some tests for this, unfortunately i had a block on getting sample OTRS data from an endpoint and don't have an API key. Is there somewhere that i can sign up at the znuny demo site? 

```
test/integration/otrs_import_test.rb:7:in `<class:OtrsImportTest>': ERROR: Need IMPORT_OTRS_ENDPOINT - hint IMPORT_OTRS_ENDPOINT='http://vz305.demo.znuny.com/otrs/public.pl?Action=ZammadMigrator' (RuntimeError)
```

**Comment by muhammadn** [comment]
Ok nevermind. got it. hints. :)

**Comment by thorsteneckel** [comment]
Hi @muhammadn - Great to have you back 👋 You can check our RSpec tests for OTRS here: https://github.com/zammad/zammad/blob/develop/spec/lib/import/otrs/user_spec.rb

**Comment by muhammadn** [comment]
@thorsteneckel so no more minitests and all new tests would be rspec?

**Comment by thorsteneckel** [comment]
Yes 🎉

**Comment by thorsteneckel** [comment]
Fixed thanks to the pull request #1677 by @muhammadn 🚀
